uint32 id
uint8 row
uint8 xpos
uint8 ypos

